wgpu-hal/src/dx12/device.rs

```diff
@@ -1052,6 +1052,7 @@ impl crate::Device<super::Api> for super::Device {
                     // FXC doesn't support SM 6.0
                     None => hlsl::ShaderModel::V5_1,
                 },
+                flags: hlsl::WriterFlags::default(),
```

Comment:
Does this enable printf at all?

Reply:
Good catch, I forgot to implement the feature logic for DX12